src/hardware/innovation.cpp

```diff
@@ -87,36 +86,51 @@ void Innovation::Open(const std::string &model_choice,
 	                                             ChannelFeature::ChorusSend,
 	                                             ChannelFeature::Synthesizer});
 
+	if (!mixer_channel->TryParseAndSetCustomFilter(channel_filter_choice)) {
+		if (channel_filter_choice != "off")
+			LOG_WARNING("INNOVATION: Invalid 'innovation_filter' value: '%s', using 'off'",
+			            channel_filter_choice.c_str());
+
+		mixer_channel->SetHighPassFilter(FilterState::Off);
+		mixer_channel->SetLowPassFilter(FilterState::Off);
+	}
+
 	const auto frame_rate_hz = mixer_channel->GetSampleRate();
 
 	// Determine the passband frequency, which is capped at 90% of Nyquist.
 	const double passband = 0.9 * frame_rate_hz / 2;
 
 	// Assign the sampling parameters
-	sid_service->setSamplingParameters(chip_clock, reSIDfp::RESAMPLE,
-	                                   frame_rate_hz, passband);
+	sid_service->setSamplingParameters(chip_clock,
+	                                   reSIDfp::RESAMPLE,
+	                                   frame_rate_hz,
+	                                   passband);
 
 	// Setup and assign the port address
 	const auto read_from = std::bind(&Innovation::ReadFromPort, this, _1, _2);
 	const auto write_to = std::bind(&Innovation::WriteToPort, this, _1, _2, _3);
-	base_port = check_cast<io_port_t>(port_choice);
+	base_port           = check_cast<io_port_t>(port_choice);
 	read_handler.Install(base_port, read_from, io_width_t::byte, 0x20);
 	write_handler.Install(base_port, write_to, io_width_t::byte, 0x20);
 
 	// Move the locals into members
 	service = std::move(sid_service);
-	channel = std::move(mixer_channel);
+	channel = mixer_channel;
```

Comment:
Suggest un-const'ing the local `mixer_channel` and keeping `std::move`.

Reply:
Done.